### ISSUE TITLE
[GUI] New UI component system

### ISSUE THREAD

**Issue by andersevenrud** [os-js/OS.js#628]
For ~~v2.2~~ *v3* I want to make use of a third-party UI component system.

As for what component system to use... that's a hard one. The existing system should be pretty easy to convert as it is sort of works like a `jQuery UI Widget` (Some plain HTML, CSS and a set of common methods).

*I'd love some input on this, as I'm only familiar with React and Vue*.

**Comment by mistergibson** [comment]
Choo might be a good candidate. I went to a local meeting where they highlighted the difference in ease of use with React. Link: https://github.com/choojs/choo

**Comment by andersevenrud** [comment]
[Moon](http://moonjs.ga/docs/getting-started.html) also looks great.

**Comment by andersevenrud** [comment]
* https://dio.js.org/
* https://hyperapp.js.org/

**Comment by salamynder** [comment]
https://mithriljs.org

**Comment by andersevenrud** [comment]
@salamynder Thank you for the suggestion :+1:

**Comment by taigrr** [comment]
I'm using Quasar right now for material themes. It's Vue-based.

**Comment by andersevenrud** [comment]
@taigrr I've been looking at Quasar a bit at work. So far I'm very impressed! I already know a bit of Vue, so it has been a very pleasant experience :)

**Comment by taigrr** [comment]
@andersevenrud Yes, I saw you mentioned Vue so I thought Quasar deserved a mention. Ionic is it's Angular counterpart.

**Comment by andersevenrud** [comment]
@taigrr Some of the Mobile app guys at work has been using Ionic for a while and seem to like it a lot. I've never actually tried it myself, but I probably will. Never hurts to know yet another framework, heh :)

**Comment by ThiccCat** [comment]
Could use https://riotjs.com

**Comment by andersevenrud** [comment]
Looks cool. Thanks for the suggestion :)

9. des. 2017 17:07 skrev "Cat" <notifications@github.com>:

Could use https://riotjs.com

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/os-js/OS.js/issues/628#issuecomment-350481846>, or mute
the thread

**Comment by junland** [comment]
Riot would be pretty cool. Its pretty lightweight too.

**Comment by mtuchi** [comment]
Sorry, i am new to this framework
Is there a boilerplate example for developing package using reactjs?

**Comment by andersevenrud** [comment]
I've tried out all suggestions, and I've settled on using Hyperapp by default. The first reason is pretty obvious: It's insanely lightweight! But what I liked the most is the functional approach (similar to Elm) and how the state is managed.

Coming from a background with Vue and React, I think this hits a nice spot between light and functionality. Hyperapp supports JSX as well, so this will be a choice.

Also, this will now come in v3 instead as this is where I focus most of my time now (and very close to a demo). I have some *very* exciting news about that coming soon... so, look out for an announcement! :tada:
